spec/models/expense_spec.rb

```diff
@@ -0,0 +1,15 @@
+require 'rails_helper'
+
+RSpec.describe Expense, type: :model do
+  subject { described_class.new(name: 'Expense Name', icon: 'GROCERY.png') }
+
+  context 'validations' do
+    it 'should have a valid name' do
+      expect(subject.name).to be_present
+    end
+
+    it 'should have a valid icon' do
+      expect(subject.icon).to be_present
+    end
+  end
+end
```

Comment:
- [ ] I can only see model specs in your project. kindly consider adding controllers (requests) and integration specs as well as per requirements.